schema/msg/msg_test.go

```diff
@@ -74,6 +74,105 @@ func TestWriteReadPointMsgWithoutOrg(t *testing.T) {
 	}
 }
 
+func TestWriteReadPointMsgArray(t *testing.T) {
+	mp := []schema.MetricPoint{
+		{
+			MKey: schema.MKey{
+				Org: 123,
+			},
+			Time:  math.MaxUint32,
+			Value: 123.45,
+		},
+		{
+			MKey: schema.MKey{
+				Org: 1,
+			},
+			Time:  math.MaxUint32,
+			Value: 1,
+		},
+		{
+			MKey: schema.MKey{
+				Org: 2,
+			},
+			Time:  math.MaxUint32,
+			Value: 12,
+		},
+	}
+	buf := make([]byte, 0, 2+32*len(mp))
+	out, err := WritePointMsgArray(mp, buf, FormatMetricPoint)
+	if err != nil {
+		t.Fatalf("%s", err.Error())
+	}
+
+	_, ok := IsPointMsg(out)
+	if !ok {
+		t.Fatal("IsPointMsg: exp true, got false")
+	}
+
+	var outPoints []schema.MetricPoint
+	iter := NewMetricPointIter(out, 6)
+	for iter.Next() {
+		outPoints = append(outPoints, iter.Value())
+	}
+
+	if iter.Err() != nil {
+		t.Fatalf("%s", iter.Err().Error())
+	}
+
+	if !reflect.DeepEqual(mp, outPoints) {
```

Comment:
for this test and the below one, i suggest making a deep copy of the input slice before passing it into any code under test.
That way, when we compare, we know we compare against a slice that cannot have been mistakenly modified by the code under test